sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -937,6 +980,15 @@ case class ExprChecksImpl(contexts: Map[ExpressionContext, ContextChecks])
         (expContext, check.support(dataType))
     }
   }
+
+  override def tagAst(exprMeta: BaseExprMeta[_]): Unit = {
+    val checks = contexts.get(AstExprContext)
+    if (checks.isEmpty) {
+      exprMeta.willNotWorkInAst(s"this is not supported by AST")
```

Comment:
Nit: Sometimes we use "this is not supported by AST" and sometimes "AST is not supported".  We should be consistent.